src/openfermion/utils/_meanfield.py

```diff
@@ -0,0 +1,127 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""This module contains constructions for mean-field Hamiltonian models.
+
+The mean-field d-wave model has the form
+
+H = - tunneling sum_{<i,j>} sum_sigma (a^dagger_{i, sigma} a_{j, sigma}
+  + a^dagger_{j, sigma} a_{i, sigma})
+  - sum_{<i,j>} Delta_{ij}
+  (a^dagger_{i, up} a^dagger_{j, down} - a^dagger_{i, down} a^dagger_{j, up}
+  + a_{j, down} a_{i, up} - a_{j, up} a_{i, down})
+
+where Delta_{ij} = +sc_gap/2 for horizontal edges and -sc_gap/2 for vertical
+edges.
+
+There are N sites and 2*N spin-orbitals. The operators a^dagger_i and a_i are
+fermionic creation and annihilation operators. One can transform these
+operators to qubit operator using the Jordan-Wigner transmation:
+
+a^dagger_j = 0.5 (X - i Y) prod_{k = 1}^{j - 1} Z_k
+a_j = 0.5 (X + i Y) prod_{k = 1}^{j - 1} Z_k
+
+These Hamiltonians live on a square lattice which has dimensions of
+x_dimension by y_dimension. They can have periodic boundary conditions or not.
+"""
+from __future__ import absolute_import
+
+from openfermion.ops import (FermionOperator,
+                             hermitian_conjugated)
+
+
+# Function to return up-orbital index given orbital index.
+def up(index):
```

Comment:
Can you perhaps import these functions from the Hubbard model module instead of rewriting them? Definitely better to avoid code duplication. Since up and down are not available for import from openfermion.utils you should import them from ._hubbard.